app/src/main/java/org/systers/mentorship/viewmodels/CommentViewModel.kt

```diff
@@ -0,0 +1,156 @@
+                            }
+                            is TimeoutException -> {
+                                message = MentorshipApplication.getContext()
+                                        .getString(R.string.error_request_timed_out)
+                            }
+                            is HttpException -> {
+                                message = throwable.message()
+                            }
+                            else -> {
+                                message = MentorshipApplication.getContext()
+                                        .getString(R.string.error_something_went_wrong)
+                                Log.e(tag, throwable.localizedMessage)
+                            }
+                        }
+                        _successfulAdd.value = false
+                    }
+                    override fun onComplete() {
+                    }
+                })
+    }
+
+    /**
+     * This function helps in adds a new task to the task list
+     * @param relationId relation for which task is to be added
+     * @param createTask to serialize task description
+     */
+    @SuppressLint("CheckResult")
+    fun addComment(relationId: Int, taskId: Int, createComment: CreateComment) {
+        commentDataManager.addComment(relationId, taskId, createComment)
+                .subscribeOn(Schedulers.newThread())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeWith(object : DisposableObserver<CustomResponse>() {
+                    override fun onNext(taskListResponse: CustomResponse) {
+                        _successfulAdd.value = true
+                    }
+                    override fun onError(throwable: Throwable) {
+                        when (throwable) {
+                            is IOException -> {
+                                message = MentorshipApplication.getContext()
+                                        .getString(R.string.error_please_check_internet)
+                            }
+                            is TimeoutException -> {
+                                message = MentorshipApplication.getContext()
+                                        .getString(R.string.error_request_timed_out)
+                            }
+                            is HttpException -> {
+                                message = throwable.message()
+                            }
+                            else -> {
+                                message = MentorshipApplication.getContext()
+                                        .getString(R.string.error_something_went_wrong)
+                                Log.e(tag, throwable.localizedMessage)
+                            }
+                        }
+                        _successfulAdd.value = false
+                    }
+                    override fun onComplete() {
+                    }
+                })
+    }
+
+    fun updateComment() {}
+}
```

Comment:
Use a composite disposable to dispose the observers once the viewmodel is destroyed

Reply:
Yup, I know. Will update the code once, the architecture has been finalized.